Source/Core/VideoCommon/VertexLoader.cpp

```diff
@@ -205,98 +205,16 @@ void VertexLoader::CompileVertexTranslator()
     m_native_vtx_decl.colors[i].components = 4;
     m_native_vtx_decl.colors[i].type = VAR_UNSIGNED_BYTE;
     m_native_vtx_decl.colors[i].integer = false;
-    switch (m_VtxDesc.low.Color[i])
```

Comment:
It'd be a pain but seeing as this class already has some unit-tests, I wonder if it'd be beneficial to write some tests before removing this with the improved replacement.

Reply:
I don't feel like writing tests exhaustive enough to cover every combination of color settings; although they'd be useful to have, it seems like a fair bit of effort which I don't want to include in this PR.  The refactoring I did here is basically just converting the switch statement into two arrays, without changing the functions themselves, so I don't think there's too much I could have broken.

It's also worth noting that the regular `VertexLoader` is not actually used by anything currently (only `VertexLoaderARM64` and `VertexLoaderX64` are used in practice since Dolphin only supports X64 and ARM64); it can be used by editing `VertexLoaderBase::CreateVertexLoader` (and in particular by defining `COMPARE_VERTEXLOADERS`; I've run the existing unit tests with that enabled and they all pass and some non-exhaustive gameplay testing also showed no issues).